packages/react-components/src/components/Switch.spec.tsx

```diff
@@ -0,0 +1,15 @@
+import { fireEvent } from '@testing-library/dom';
+import * as React from 'react';
+import { render } from '../test-utils';
+import { Switch } from './Switch';
+
+describe('Switch', () => {
+  const toggleFn = jest.fn();
+
+  it('should call on toggle when clicked', () => {
```

Comment:
It would be nice to also test the proper behavior of the `Switch` component in terms of changing its states when the user interacts with it:
- When it is **enabled** -> user clicks it -> expect to be **disabled** 
- When it is **disabled** -> user clicks it -> expect to be **enabled**

Additionally, it would also be beneficial to test the duality of the Switch component nature: 
- Controlled input (changes in property `on` are directly reflected in the component visual and logical state)
- Uncontrolled input (maintain its internal state on its own and only notifies about changes)